### PR TITLE
Improvements in sd_card and spi

### PR DESCRIPTION
**Pull request by mchindri**
This drivers will be used in the smoke detector application an this improvements where needed.
More details in each commit message.

### CODE REVIEW

**Comment by mchindri** [comment]
v2:
 - Fix codacy
 - Add TIMEOUT_LOOP: check commit

**Comment by mchindri** [comment]
v3:
 - Modify commit descriptions
 - Implement timeout with delay

**Comment by mchindri** [comment]
v4:
 - split sd-card modifications in multiple commits

**Comment by mchindri** [comment]
v5:
 - solve typo in commit message

**Comment by mchindri** [comment]
v6:
 - Fix astyle

File: drivers/sd-card/sd.c [hunk 1]
```diff
@@ -117,19 +125,23 @@ static int32_t wait_for_response(struct sd_desc *sd_desc, uint8_t *data_out)
...
-			return FAILURE;
-		if (timeout-- == 0)
-			return FAILURE;
-	}
+			break;
+		if (data != 0x00) {
+			ret = SUCCESS;
+			break;
+		}
+	} while (not_timeout--);
```
Comment (adrimbarean) [inline drivers/sd-card/sd.c]: Should we have an `mdelay` in this while too?

File: drivers/sd-card/sd.h [hunk 2]
```diff
@@ -87,7 +93,7 @@ struct sd_init_param {
  */
 struct sd_desc {
 	/** Descriptor of an initialized SPI channel */
-	struct spi_desc *spi_desc;
+	struct spi_desc	*spi_desc;
```
Comment (adrimbarean) [inline drivers/sd-card/sd.h]: Is this change wanted or just a small error?

Comment (mchindri) [reply drivers/sd-card/sd.h]: It was wanted. I replaced a space with a tab

File: drivers/sd-card/sd.c (hunk 1 above)
Comment (mchindri) [reply drivers/sd-card/sd.c]: Yes we should, I missed it

**Comment by mchindri** [comment]
v7:
 - Add missing mdelay

File: drivers/platform/aducm3029/spi.c [hunk 3]
```diff
@@ -44,56 +45,63 @@
...
 /******************************************************************************/
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 
-/**
- * @brief Allocates the memory needed for the descriptor
- * @return A address to the allocated memory, NULL if the allocation fails.
- */
-static struct spi_desc *alloc_data_mem(void)
+static int32_t config_device(struct aducm_device_desc *dev,
```
Comment (cristiantpop) [inline drivers/platform/aducm3029/spi.c]: Add function comment.

**Comment by mchindri** [comment]
v8
 - Add comment for config_device

**Comment by mchindri** [comment]
Will merge if no more comments today.

**Pull request merged**